topic-controller/src/test/java/io/strimzi/test/Oc.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.strimzi.test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Oc extends KubeCluster {
```

Comment:
I have a doubt about this `Oc` name. We are going to spin an OpenShift cluster using `oc` tool. The other class is called `Minikube`, not `Kubectl` :-) Maybe we could have another class named `Minishift` if we need to spin the OpenShift cluster in that way but I would call this class just `OpenShiftCluster` ... but it's just my opinion :-)

Reply:
>  The other class is called `Minikube`, not `Kubectl`

It's started using `minikube`, even some of the methods use `kubectl`. I didn't add a `Minishift` because I didn't need one right now. If and when we need a `Minishift` I think there would be an opportunity to refactor because it would have a lot in common with `Minikube`. Probably the same class with static factory methods for each kind.

Reply:
But what about renaming `Oc` to something like `OpenShiftCluster`. At the end if I think more on that `Oc` is extending `KubeCluster` but oc is a "cluster" itself ... it's the tool. It's a nit I know ... but the oc name for this class isn't clear to me.

Reply:
Well this is interesting (and perhaps it exposes my ignorance). `oc` is, on the one hand the OpenShift client. But running `oc cluster up` is not the same thing as `minishift start`. But minishift is also OpenShift, isn't it? So can I do a `minishift start` and then use `oc` with that cluster? Presumably I can also use `kubectle` with an OpenShift Cluster? So logically, perhaps I need to separate the starting of a cluster from the tool I use to interact with it once it's running. Conceptually woudl that make more sense? Since both `Oc` and `Minikube` are stateless that wouldn't be difficult to do.

Reply:
`oc` is just the tool. The `cluster up` command just downloads an OpenShift Origin Docker image and starts this image on your local PC in order to have an OpenShift cluster running.
At this point you can use all the other `oc` commands (i.e. get pods, create, ...) for interacting with the OpenShift cluster.
`minishift start` starts a Virtual Machine and inside it does something like the `oc cluster up` does on your local PC do downloading Docker image and starting it : at this point the OpenShift cluster is running in a VM (which is your single cluster node, while in the previous case it's your PC). With `minishift` you have a out-of-box environment and OpenShift cluster without dirtying your PC because all is downloaded and run in a VM.

> Presumably I can also use kubectle with an OpenShift Cluster?

Yes for the common features between Kube and OpenShift

What I'm suggesting is just changing the name to `Oc` into something more related to a "cluster" name. `minikube` is a tool but finally it's a cluster :-) for this reason I like having `Minikube` derived from `KubeCluster` but not `Oc`

Reply:
But as I said it's a nit ... more "naming" related not so important .. I can live with `Oc` ;)

Reply:
I have refactored this into `KubeCluster` hierarchy (`OpenShift`, `Minikube` subclasses)  and a `KubeClient` hierarchy (`Oc` and `Kubectl` subclasses) and I think it's nicer. It will allow us to expose the client through the `KubeClusterResource`, so that, for example, individual tests can interact with the cluster in their own way. Thanks for pushing me into doing this @ppatierno!